contracts/src/PDPVerifier.sol

```diff
@@ -344,16 +353,22 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
     // proof set Merkle roots at some epoch. The challenge seed is determined
     // by the epoch of the previous proof of possession.
     // Note that this method is not restricted to the proof set owner.
-    function provePossession(uint256 setId, Proof[] calldata proofs) public {
+    function provePossession(uint256 setId, Proof[] calldata proofs) public payable{
         uint256 challengeEpoch = nextChallengeEpoch[setId];
         require(block.number >= challengeEpoch, "premature proof");
         require(proofs.length > 0, "empty proof");
-        uint256 seed = drawChallengeSeed(setId);
 
+        // Calculate and burn the proof fee
+        uint256 proofFee = PDPFees.proofFee(proofs.length);
+        burnFee(proofFee);
+        if (msg.value > proofFee) {
+            // Return the overpayment
+            payable(msg.sender).transfer(msg.value - proofFee);
+        }
```

Comment:
Can you extract the "burn and return change" logic into a function for re-use? Directly into `burnFee` is probably enough.